src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -159,51 +184,94 @@ public static List<int> CommonBaudRates()
             };
         }
 
+        /// <summary>
+        /// Adds a new command handler.
+        /// A command handler can support extended commands.
+        /// </summary>
+        /// <typeparam name="T">An instance of <see cref="ExtendedCommandHandler"/>.</typeparam>
+        /// <param name="newCommandHandler">The new handler</param>
+        public void AddCommandHandler<T>(T newCommandHandler)
+            where T : ExtendedCommandHandler
+        {
+            if (newCommandHandler == null)
+            {
+                throw new ArgumentNullException(nameof(newCommandHandler));
+            }
+
+            _commandHandlersLock.EnterWriteLock();
+            try
+            {
+                if (newCommandHandler.HandlesMode != null)
+                {
+                    // If we already know the mode, replace its configuration with the new one (typically, this will only update the name)
+                    var m = _knownSupportedModes.FirstOrDefault(x => x.Value == newCommandHandler.HandlesMode.Value);
+                    if (m != null)
+                    {
+                        _knownSupportedModes.Remove(m);
+                    }
+
+                    _knownSupportedModes.Add(newCommandHandler.HandlesMode);
+
+                    // Update internal list
+                    Firmata.SupportedModes = _knownSupportedModes;
+                }
+
+                _extendedCommandHandlers.Add(newCommandHandler);
+            }
+            finally
+            {
+                _commandHandlersLock.ExitWriteLock();
+            }
+
+            if (_firmata != null)
+            {
+                // Only if already initialized
+                newCommandHandler.Registered(_firmata, this);
+            }
+        }
+
+        /// <summary>
+        /// Gets the command handler with the provided type. An exact type match is performed.
+        /// </summary>
+        /// <typeparam name="T">The type to query</typeparam>
+        /// <returns>The command handler, or null if none was found</returns>
+        public T? GetCommandHandler<T>()
```

Comment:
NIT: not sure what the use case for this is, but in case you want to support type hirerarchy as well, you can use Type.IsAssignableFrom

Reply:
No, see krwq's comment below. This method is designed as a "service query". You ask it for a specific interface (i.e. whether DHT support is available). Asking for base classes doesn't make sense (and wouldn't work, because this only returns one instance at a time, not several).